modules/ding_place2book/ding_place2book.module

```diff
@@ -545,20 +550,27 @@ function ding_place2book_get_event_state($event_maker_id, $event_id) {
           ];
         }
 
+        // Find the closest sale period.
+        $begin_values = array_column($sale_periods, 'begin');
+        $min_period_begin = min($begin_values);
```

Comment:
This selects the lowest/first sales period. Is there a risk that one sales period is in the past (and thus over) while there may be another current open one?

Example:
```php
$sales_periods = [
  [ 'begin' => 1554076800, 'end' => 1556582400 ], // <-- this should be selected through April 2019
  [ 'begin' => 0, 'end' => 1 ], // <-- this would be selected
];

Reply:
Yes, there is chance for this to happen.

I would also like to note that if we find an active sales periods there's really no need to continue with all this. The status will just be sales open. Hence my suggestion to keep the original loop, which handled this pretty well IMO.

There was a lot of gotchas when i wrote the code. I'm not sure it's a good idea to change it too much.